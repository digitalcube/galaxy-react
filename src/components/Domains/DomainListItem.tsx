import React, { FC, PropsWithChildren } from 'react';
import { DomainListItemTitle } from './DomainListItemTitle';

export type DomainListItemProps = PropsWithChildren<{
  domainName: string;
}>;
export const DomainListItem: FC<DomainListItemProps> = ({
  domainName,
  children,
}) => {
  return (
    <div className="domain-list-item mt-2 px-3 py-4 rounded d-flex align-items-center justify-content-between">
      {domainName ? (
        <DomainListItemTitle>{domainName}</DomainListItemTitle>
      ) : null}
      <div className="domain-misc d-flex align-items-center">{children}</div>
    </div>
  );
};
