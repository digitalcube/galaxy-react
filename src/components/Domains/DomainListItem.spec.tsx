/* eslint-disable jsx-a11y/anchor-is-valid */
import { render } from '@testing-library/react';
import React from 'react';
import { DomainListItem } from './DomainListItem';

describe('DomainListItem', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(
      <DomainListItem domainName="example.com">
        <a href="#">Item</a>
      </DomainListItem>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
