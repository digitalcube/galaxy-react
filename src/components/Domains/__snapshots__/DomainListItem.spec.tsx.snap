// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainListItem to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="domain-list-item mt-2 px-3 py-4 rounded d-flex align-items-center justify-content-between"
  >
    <div
      class="domain-name font-weight-bold"
    >
      example.com
    </div>
    <div
      class="domain-misc d-flex align-items-center"
    >
      <a
        href="#"
      >
        Item
      </a>
    </div>
  </div>
</DocumentFragment>
`;
